djangocms_link/admin.py

```diff
@@ -0,0 +1,215 @@
+from django.apps import apps
+from django.conf import settings
+from django.contrib import admin
+from django.core.exceptions import FieldError, PermissionDenied
+from django.db.models import OuterRef, Q, Subquery
+from django.http import JsonResponse
+from django.urls import path
+from django.views.generic.list import BaseListView
+
+from cms import __version__
+from cms.models import Page
+from cms.utils import get_language_from_request
+
+from . import models
+from .helpers import get_manager
+
+
+_version = int(__version__.split(".")[0])
+if _version >= 4:
+    from cms.admin.utils import GrouperModelAdmin
+    from cms.models import PageContent
+else:
+    from cms.models import Title as PageContent
+
+    class GrouperModelAdmin:
+        pass
+
+REGISTERED_ADMIN = getattr(settings, "DJANGOCMS_LINKABLE_MODELS", "auto")
```

Comment:
`DJANGOCMS_LINKABLE_MODELS` is either "auto" (default) for autodetection or a list of strings in the form of `"<app_label>.<model_name>"` and controls which models will be selectable from the autocomplete field besides pages. 

Example: `DJANGOCMS_LINKABLE_MODELS = ["djangocms_blog.post"]` will (only) add blog posts to the selection (besides pages). 

`DJANGOCMS_LINKABLE_MODELS = []` will ensure only pages are offered.

It is evaluated in the app's `ready()` method to ensure all models are registered and all model admins are available.

Reply:
Since the heavy lifting is done by django's admin, the internal `REGISTERED_ADMIN` is converted to a list of model admin classes (which contain the model in the `model` attribute). 

Maybe it is less confusing if the `ready()` method reads the setting, and `REGISTERED_ADMIN ` is initialized empty?

Reply:
Thanks for taking the time to dive deep! 🚀